pkg/cloud/services/compute/instance.go

```diff
@@ -126,11 +127,23 @@ func (s *Service) constructNetworks(openStackCluster *infrav1.OpenStackCluster,
 	return nets, nil
 }
 
-func (s *Service) CreateInstance(eventObject runtime.Object, openStackCluster *infrav1.OpenStackCluster, instanceSpec *InstanceSpec, clusterName string) (*InstanceStatus, error) {
-	return s.createInstanceImpl(eventObject, openStackCluster, instanceSpec, clusterName, retryIntervalInstanceStatus)
+func (s *Service) CreateInstance(eventObject runtime.Object, openStackCluster *infrav1.OpenStackCluster, instanceSpec *InstanceSpec, clusterName string, isBastion bool) (*InstanceStatus, error) {
+	return s.createInstanceImpl(eventObject, openStackCluster, instanceSpec, clusterName, isBastion, retryIntervalInstanceStatus)
 }
 
-func (s *Service) createInstanceImpl(eventObject runtime.Object, openStackCluster *infrav1.OpenStackCluster, instanceSpec *InstanceSpec, clusterName string, retryInterval time.Duration) (*InstanceStatus, error) {
+func (s *Service) getAndValidateFlavor(flavorName string, isBastion bool) (*flavors.Flavor, error) {
+	f, err := s.getComputeClient().GetFlavorFromName(flavorName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting flavor from flavor name %s: %v", flavorName, err)
+	}
+	if !isBastion && f.VCPUs <= 1 {
+		return nil, fmt.Errorf("error create machines with flavor's vCPU less than 2, this will cause kubeadm fail to start, update your flavor setting to make VCPUS equal to or bigger than 2")
```

Comment:
I found this a bit hard to understand, would suggest something like this instead:
```suggestion
		return nil, fmt.Errorf("kubeadm requires a minimum of 2 vCPUs, pick a flavor with at least 2 vCPUs")
```

Reply:
ok, will update